generic/tests/nfs_corrupt.py

```diff
@@ -64,14 +72,31 @@ def _get_service_cmds(self):
 
         return [start_cmd, stop_cmd, restart_cmd, status_cmd]
 
-    @error.context_aware
+    @error_context.context_aware
     def setup(self, force_start=False):
         """
         Setup test NFS share.
 
         :param force_start: Whether to make NFS service start anyway.
         """
-        error.context("Setting up test NFS share")
+        error_context.context("Checking available space to export",
+                              logging.info)
+        tmpdir = os.path.split(self.nfs_dir)[0]
```

Comment:
for 'df', it's the same result no matter get the tmpdir.

Reply:
actually, it matters since self.nfs_dir is not created yet at this point.